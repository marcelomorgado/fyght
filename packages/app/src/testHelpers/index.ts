import { BigNumber } from "ethers";
import { Skin } from "../constants";

export const storeMocks: { myFyghter: Fyghter; enemies: Enemy[] } = {
  myFyghter: {
    id: BigNumber.from("1"),
    name: "John",
    skin: Skin.NORMAL_GUY,
    xp: BigNumber.from("5"),
    balance: BigNumber.from("5"),
  },
  enemies: [
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("2"),
        name: "Charlie",
        skin: Skin.NAKED,
        xp: BigNumber.from("1"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("3"),
        name: "Saul",
        skin: Skin.NINJA,
        xp: BigNumber.from("10"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("4"),
        name: "Julian",
        skin: Skin.NAKED,
        xp: BigNumber.from("1"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("5"),
        name: "Brad",
        skin: Skin.MASTER,
        xp: BigNumber.from("89"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("6"),
        name: "Frank",
        skin: Skin.NO_ONE,
        xp: BigNumber.from("71"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("7"),
        name: "Brian",
        skin: Skin.MONK,
        xp: BigNumber.from("30"),
        balance: BigNumber.from("5"),
      },
    },
    {
      winProbability: BigNumber.from("50"),
      fyghter: {
        id: BigNumber.from("8"),
        name: "Fred",
        skin: Skin.KARATE_KID,
        xp: BigNumber.from("10"),
        balance: BigNumber.from("5"),
      },
    },
  ],
};
