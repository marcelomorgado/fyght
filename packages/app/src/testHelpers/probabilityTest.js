const p = (myXp, hisXp) => 1000 / (myXp + hisXp);

for (let myXp = 1; myXp <= 100; ++myXp) {
  for (let hisXp = 1; hisXp <= 100; ++hisXp) {
    console.log(`myXp=${myXp}, hisXp=${hisXp}, p=${p(myXp, hisXp)}`);
  }
}
