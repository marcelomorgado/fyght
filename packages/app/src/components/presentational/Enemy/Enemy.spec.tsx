import * as React from "react";
import { shallow } from "enzyme";
import { storeMocks } from "../../../testHelpers";
import { Enemy } from ".";

const { enemies, myFyghter } = storeMocks;

describe("Enemy", () => {
  test("should render the component", () => {
    const [enemy] = enemies;
    expect(shallow(<Enemy enemy={enemy} myFyghter={myFyghter} />)).toMatchSnapshot();
  });
});
