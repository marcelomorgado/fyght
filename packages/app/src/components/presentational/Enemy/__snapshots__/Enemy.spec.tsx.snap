// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enemy should render the component 1`] = `
<Card
  hoverable={true}
  title="Charlie"
  type="inner"
>
  <Component
    size="medium"
    skin="naked"
  />
  <p />
  <p>
    XP: 1
  </p>
  <p>
    Balance: 
    <Component
      gainIfWin={
        BigNumber {
          "_hex": "0x4563918244f3ff06",
        }
      }
      value={
        BigNumber {
          "_hex": "0x05",
        }
      }
    />
  </p>
  <p>
    Win probability: 
    <WinProbability
      value={
        BigNumber {
          "_hex": "0x32",
        }
      }
    />
  </p>
  <p>
    Gain if win: 
    <GainOrLoss
      value={
        BigNumber {
          "_hex": "0x4563918244f3ff06",
        }
      }
    />
  </p>
  <p>
    Loss if lose: 
    <GainOrLoss
      value={
        BigNumber {
          "_hex": "-0xfa",
        }
      }
    />
  </p>
  <Component
    disabled={true}
    enemyId={
      BigNumber {
        "_hex": "0x02",
      }
    }
  />
  <p />
  <Alert
    message="Your fyghter with insufficient funds."
    type="warning"
  />
</Card>
`;
